lib/watcher.js

```diff
@@ -113,6 +126,30 @@ Watcher.prototype.updateTestDependencies = function (file, sources) {
 	}
 };
 
+Watcher.prototype.trackExclusivity = function (api) {
+	var self = this;
+	api.on('stats', function (stats) {
+		self.updateExlusivity(stats.file, stats.hasExclusive);
+	});
+};
+
+Watcher.prototype.updateExlusivity = function (file, hasExclusive) {
+	var ix = this.exclusiveTests.indexOf(file);
+
+	if (hasExclusive && ix === -1) {
+		this.exclusiveTests.push(file);
+	} else if (!hasExclusive && ix !== -1) {
+		this.exclusiveTests.splice(ix, 1);
```

Comment:
We're splicing out, actually. It's just a weird API. It seems fairly clear to me that one branch adds the file, if necessary, and the other removes, if necessary.

Could use the `union()` and `diff()` helpers if that would make it clearer?